styx-common/src/main/java/com/spotify/styx/util/TimeUtil.java

```diff
@@ -210,36 +214,33 @@ public static boolean isAligned(Instant instant, Schedule schedule) {
    * @return A zoned date time with the offset applied
    */
   public static ZonedDateTime addOffset(ZonedDateTime time, String offset) {
-    final int tPos = offset.indexOf('T');
-    final String periodOffset;
-    final String durationOffset;
+    final Pattern pattern = Pattern.compile("([-+]?)P([-+0-9YMWD]+)?(T([-+0-9HMS.,]+)?)?",
```

Comment:
Would be nice to not have to use regex but if we have to, at least make the compiled pattern a `static final` constant